libsolidity/ast/AST.cpp

```diff
@@ -338,8 +338,14 @@ TypePointer FunctionDefinition::type() const
 TypePointer FunctionDefinition::typeViaContractName() const
 {
 	if (annotation().contract->isLibrary())
-		return FunctionType(*this).asCallableFunction(true);
-	return TypeProvider::function(*this, FunctionType::Kind::Declaration);
+	{
+		if (isPublic())
+			return FunctionType(*this).asCallableFunction(true);
+		else
+			return TypeProvider::function(*this, FunctionType::Kind::Internal);
```

Comment:
One line less, but not sure which is more correct according to coding style
```suggestion
		return isPublic() ?
			FunctionType(*this).asCallableFunction(true) : 
			TypeProvider::function(*this, FunctionType::Kind::Internal);
```

Reply:
Your suggestion would need a line break after `return` and I think the ternary operator is in general harder to understand :)